### DISCUSSION TITLE
Integrating python scripts in the application and returning complex data

### DISCUSSION THREAD

**Discussion by di-git-ali** [tauri-apps/tauri#6018] [Q&A]
### Question 💭 
Hello! We are building a small program that allows image processing and video manipulation, and we are trying to integrate some python scripts into the tauri application.
It seems that the only way to communicate with python is to use it as a sidecar and get the data via `stdout` by printing the retrieved output in the python functions, which is not very convenient for our use case. 
Is there a better way to do this? It seems that our approach is wrong, and we are not sure how to find a neater solution.

Thanks in advance!

**Comment by FabianLars** [comment]
> Is there a better way to do this?

If you don't want to use rust, not really no. Either stdio or an embedded localhost server on the python side :/
If you _can_ use rust there are other IPC option though i'm not sure what works in python. Thinking about named pipes and stuff like that.

To bundle the scripts themselves you always need a tool like pyinstaller or pyoxidizer, unless it's fine to require a python installation on the user's system. Or if you're adventurous there's stuff like https://github.com/RustPython/RustPython, but including dependencies for image processing may be really hard, idk

**Reply by DennisZhangOiler** [reply]
> > Is there a better way to do this?
> 
> If you don't want to use rust, not really no. Either stdio or an embedded localhost server on the python side :/ If you _can_ use rust there are other IPC option though i'm not sure what works in python. Thinking about named pipes and stuff like that.
> 
> To bundle the scripts themselves you always need a tool like pyinstaller or pyoxidizer, unless it's fine to require a python installation on the user's system. Or if you're adventurous there's stuff like https://github.com/RustPython/RustPython, but including dependencies for image processing may be really hard, idk

@FabianLars Hi, how to include a python installation when build on windows and distribute as windows installer?
